Щур Всеволод

VsevolodShchur opened the pull request.

VsevolodShchur committed:
Щур Всеволод

goverdovskiy commented on TagsCloudContainer/AutofacConfig.cs:
```diff
@@ -0,0 +1,110 @@
+                .Then(settings => settings.WithDebugMarkingColor(debugMarkingColorResult.GetValueOrThrow()))
+                .Then(settings => settings.WithDebugItemRectangleColor(debugItemRectangleColorResult.GetValueOrThrow()))
+                .Then(settings => settings.WithSize(imageSizeResult.GetValueOrThrow()))
+                .RefineError("Failed configuring settings");
```
Как то много GetValueOrThrow, и такие места есть еще. Надо подумать, как не спамить GetValueOrThrow в каждой строке.

goverdovskiy commented on TagsCloudContainer/WordProcessing/ConsoleReader.cs:
```diff
@@ -0,0 +1,23 @@
+            var words = new HashSet<string>();
+            while (true)
+            {
+                var word = Console.ReadLine();
```
Надо хотя бы писать, что ожидаются слова, а то на вид будто приложение зависло.

goverdovskiy commented:
Не во все потенциально опасные интерфейсы возвращают `Result<T>`, например `IImageSaver`.
Кажется, `Result<T>` не везде обязателен, например в приватных методах его возможно использовать поменьше, т.к. значительно усложняется дебаг.
`Result<T>` хорошо подходит для публичных методов, чтобы дать пользователю чуть больше свободы и надежности при вызове метода, а дальше он без `try catch` сможет обработать ошибку.

VsevolodShchur committed:
ЩВ2

-Недостающие Result'ы
-Выводится сообщение в ConsoleReader
-Избавился от спама GetValueOrThrow
-Добавил файл с настройками
